### DISCUSSION TITLE
Ive made a social media site with pocketbase x pockethost using vanilla js - it is really slow but when it works its amazing

### DISCUSSION THREAD

**Discussion by MalikWhitten67** [pockethost/pockethost#166] [Show and tell]
The project took me a total of 13 days - features: image & video & text posting, videos can only be 6mb in size min quality is 720p 30
Ive added keywords for emojis like grinning_emoji t- and embedding youtube videos for people who want bigger upload limits. ive allowed certain html tags in posts for those who want to use them to organize post content like making articles etc.

Framework I used: None - this is static html & regular js :}
![image](https://user-images.githubusercontent.com/65188863/213580861-ade519d3-aca6-4b7d-beb6-183b64b2e86b.png)
![image](https://user-images.githubusercontent.com/65188863/213580943-a924ad25-54cb-4d72-99ab-b80becaeaa19.png)

https://user-images.githubusercontent.com/65188863/213581046-272d6cef-b3df-4d7c-a6ee-9ecab9e1045e.mp4

**Comment by MalikWhitten67** [comment]
Url: https://www.postr-inc.me -  mobile only currently - pc soon! fixed it should work on any phone now

**Reply by benallfree** [reply]
It's loading a blank page for me :(

**Comment by benallfree** [comment]
@MalikWhitten67 Congratulations!! When we get the Showcase feature working https://github.com/benallfree/pockethost/issues/167

> it is really slow

What is slow?

**Reply by MalikWhitten67** [reply]
So when requesting data it's around 300ms. Which slows down time for dom to finish loading since it has to wait which totals to 700ms before they can see the page

**Reply by MalikWhitten67** [reply]
Yes as mentioned some phones do not work - as I'm still experimenting with different ones - the ones that are stable are any phone withing 360 no 375 - 414 x 915 if you are on PC you can just go into dev tools and select s20 to see the page!

**Reply by benallfree** [reply]
@MalikWhitten67 Sounds like you can make some optimizations so the page load doesn't block awaiting data.

Is there a reason you're not using a framework that handles all of these issues already? Something like SvelteKit?

**Reply by MalikWhitten67** [reply]
Well, I do skill tests often - which when I do them, I do not use frameworks. And to be fair I prefer vanilla over new technology - I only use frameworks when working alongside others but personal projects that likely will not be scaled any time soon no.

**Reply by benallfree** [reply]
@MalikWhitten67 Okay that makes sense. Of course we will want to showcase things that broadly work for people. It wouldn't make sense to have a showcase link to a site that didn't load :)

**Reply by MalikWhitten67** [reply]
yeah true

**Comment by MalikWhitten67** [comment]
![image](https://user-images.githubusercontent.com/65188863/215372295-f5b79d57-8e5a-436b-984f-d5485e72ffee.png)

I ended up switching to astro.js and it fixed all my issues with it being very slow. I am now redoing the whole UI design, which I think I am doing better than before. This is the new landing page, which I think I have done very well. Every time I tried making a landing page it ended up being terrible, but this is probably my best one.
